packages/store-sync/src/createPreconfirmedBlockStream.ts

```diff
@@ -81,13 +85,15 @@ export function createPreconfirmedBlockStream(opts: PreconfirmedBlockStreamOptio
         catchError((e) => {
           debug("Error in preconfirmed logs stream, recreating", e);
           recreatePreconfirmedStream$.next();
-          return throwError(() => e);
+          return of(null);
```

Comment:
Throwing an error here in combination with the delay in recreating caused the whole `createPreconfirmedBlockStream` to be closed and to fall back to the regular sync in `createStoreSync`. It's also not necessary to throw the error here since the expectation is we're reconnecting the stream.